prov/efa/src/rxr/rxr_pkt_type_data.c

```diff
@@ -61,11 +62,21 @@ ssize_t rxr_pkt_send_data(struct rxr_ep *ep,
 	data_pkt = (struct rxr_data_pkt *)pkt_entry->pkt;
 	data_pkt->hdr.seg_size = payload_size;
 
-	pkt_entry->pkt_size = ofi_copy_from_iov(data_pkt->data,
-						payload_size,
-						tx_entry->iov,
-						tx_entry->iov_count,
-						tx_entry->bytes_sent);
+#ifdef HAVE_LIBCUDA
+	if (rxr_ep_is_cuda_mr(tx_entry->desc[0]))
+		pkt_entry->pkt_size = ofi_copy_from_cuda_iov(data_pkt->data,
+							     payload_size,
+							     tx_entry->iov,
+							     tx_entry->iov_count,
+							     tx_entry->bytes_sent);
+       else
+#endif
+		pkt_entry->pkt_size = ofi_copy_from_iov(data_pkt->data,
+							payload_size,
+							tx_entry->iov,
+							tx_entry->iov_count,
+							tx_entry->bytes_sent);
+
```

Comment:
Can we encapsulate this section into one function? It has been used in quite a few places, and might worth the effort.

Reply:
Yep, can do that. I thought of updating the util in `ofi_iov.h` to do this based off the iov buffer type so all providers could use it, but that needed more work since an iov could potentially mixed. Will combine these into an EFA provider util for now.